crates/bevy_ecs/src/relationship/mod.rs

```diff
@@ -286,14 +364,22 @@ pub trait RelationshipTarget: Component<Mutability = Mutable> + Sized {
         let (entities, mut commands) = world.entities_and_commands();
         let relationship_target = entities.get(entity).unwrap().get::<Self>().unwrap();
         for source_entity in relationship_target.iter() {
-            commands.entity(source_entity).try_despawn();
+            let command = move |source_entity_mut: EntityWorldMut| {
+                if let Some(relationship) = source_entity_mut.get::<Self::Relationship>() {
+                    // Only despawn if this source entity's target collection only contains the entity being despawned
+                    if relationship.len() == 1 && relationship.get().contains(entity) {
+                        source_entity_mut.despawn();
```

Comment:
I know this is unrelated to your PR, but I don't understand this part of the code anymore. 
The docs say that Relationship is the source of truth; so how come we despawn the source entity if it has no more targets? Shouldn't we at least check for LINKED_SPAWN?